system-helper/flatpak-system-helper.c

```diff
@@ -224,6 +418,44 @@ handle_deploy (FlatpakSystemHelper   *object,
       return TRUE;
     }
 
+  for (guint i = 0; i < ongoing_pulls->len; i++)
+    {
+      FlatpakSystemHelperOngoingPull *pull_tmp;
+      g_autofree gchar *user_folder = NULL;
+
+      user_folder = g_file_get_path (g_file_get_parent (path));
```

Comment:
Why not do this outside the loop? `path` doesn't change between iterations.

Consider factoring the loop out to a `get_ongoing_pull_by_dir ()` function that returns a possibly-NULL `FlatpakSystemHelperOngoingPull`

Reply:
Also, `g_file_get_parent()` returns a new ref, which is leaked here